altair/api.py

```diff
@@ -409,6 +409,70 @@ def to_dict(self):
         D['config'] = self.vlconfig.to_dict()
         return D
 
+    def select_x(self):
+        """
+        Helper function that does a best effort of selecting an automatic x axis.
+
+        Raises AssertionError if it cannot find a x axis to choose.
+        """
+        assert self.data is not None
+        assert len(self.data) >= 1
+
+        d = self._classify_data_by_type()
+
+        # Choose the first column found on the following order: T, O, N, Q
+        chosen_x = None
+        for typ in ['Q', 'N', 'O', 'T']:
+            if len(d[typ]) >= 1:
+                chosen_x = d[typ][0]
+
+        if chosen_x is None:
+            raise AssertionError('Could not select X.')
```

Comment:
I completely agree @ellisonbg. With complex datasets, there's no hope of getting it right, so I'm not making the functions very complex.

Instead, I think we should focus on creating a UI a la Polestar that will give users control over their graph. For this UI, we could choose to display a default graph as a starting point, for example.

Besides the UI, these helper methods will allow us to create something similar to what other notebooks currently have. Users are very vocal about the ability to instantly visualize something, even if it's not the right graph at first. They later ask for tweaking, but it's the instant visualization of their data that they love.

I think that the UI and these methods are a killer combination. These methods are small and simple enough that they should be easy to maintain.